src/features/applications/hooks/slug.ts

```diff
@@ -17,10 +17,14 @@ hooks.addPureHook('POST', 'resin', 'application', {
 				},
 			});
 			if (organization != null) {
-				request.values.slug = getApplicationSlug(
-					organization.handle,
-					request.values.app_name,
-				);
+				try {
+					request.values.slug = getApplicationSlug(
+						organization.handle,
+						request.values.app_name,
+					);
+				} catch (e) {
+					throw new errors.NotFoundError(e.message);
```

Comment:
:+1: 
I saw that we don't have a test case for this, not even in the SDK...
I'm fine leaving it as a 404 to preserve the current behavior, but let's add comment that this was done for backwards compatiblity.
On the other hand, I would be fine even if you changed this to a bad request, if you also had thought about doing so, since I'm not expecting that there would be users depending on the 404 status, when they pass in a null app_name.